### PR TITLE
Braintree Blue: Support Level 2 and 3 data fields

### PR DESCRIPTION
**Pull request by curiousepic**
Remote:
68 tests, 384 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Unit:
57 tests, 150 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/braintree_blue.rb [hunk 1]
```diff
@@ -656,6 +657,26 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
...
         end
 
+        parameters[:tax_amount] = options[:tax_amount] if options[:tax_amount]
+        parameters[:tax_exempt] = options[:tax_exempt] if options[:tax_exempt]
+        parameters[:purchase_order_number] = options[:purchase_order_number] if options[:purchase_order_number]
+
+        parameters[:shipping_amount] = options[:shipping_amount] if options[:shipping_amount]
+        parameters[:discount_amount] = options[:discount_amount] if options[:discount_amount]
+        parameters[:ships_from_postal_code] = options[:ships_from_postal_code] if options[:ships_from_postal_code]
+
```
Comment (molbrown) [inline lib/active_merchant/billing/gateways/braintree_blue.rb]: Are the fields `shipping.postal_code` and `shipping.country_code_alpha3` already handled elsewhere? I noticed those in the [docs ](https://developers.braintreepayments.com/reference/general/level-2-and-3-processing/required-fields/ruby#level-3-data) but don't see them built in right here.

Comment (curiousepic) [reply lib/active_merchant/billing/gateways/braintree_blue.rb]: Yep, in https://github.com/activemerchant/active_merchant/blob/35666a09c074ebc2c7e09f9cfa2cb8d1fc4c9e70/lib/active_merchant/billing/gateways/braintree_blue.rb#L318

**Comment by molbrown** [comment]
Looks like Travis has a problem with something other than these changes.

**Comment by curiousepic** [comment]
Yeah looks like the `create_transaction_parameters` is too long for Rubocop. Do feel like it should be broken down into sub methods; I'll look into that.

**Comment by curiousepic** [comment]
@molbrown Refactored `create_transaction_parameters`, give it another once-over? I'm not too concerned with the Travis failures right now.

**Comment by molbrown** [comment]
Nice, LGTM.

**Pull request closed without merging**